backend/lib/server_code/api/roomRoutes.js

```diff
@@ -14,24 +19,40 @@ roomRouter.use((req, res, next) => {
   next();
 });
 
-roomRouter.post("/create", (req, res) => {
+roomRouter.post("/create", async (req, res) => {
   const { creatorID } = req.body;
-  //roomPlayers.push(creatorID);
+  const roomID = roomIDCreator();
   roomCreated = true;
-  res.send({ roomID: 1, creatorID, creationDate: res.date });
+  let [socket, socketPort] = [null, null];
+  try {
+    [socket, socketPort] = await createNewSocket();
+    console.log(`socket port got from function:${socketPort}`);
+    socketsCollection[roomID] = [socket, socketPort];
+    res.send({
+      roomID: roomID,
+      creatorID,
+      creationDate: res.date,
+      socketPort: socketPort,
+    });
+  } catch (e) {
+    console.log(`An error occured while getting socket: ${e}`);
+  }
+
+  console.log("here!");
 });
 
 roomRouter.get("/join", (req, res) => {
```

Comment:
u need to create a socket for each player joining the room, where did you create that socket?

Reply:
We talked about it , socket is for room